packages/p2p/src/pages/advertiser-page/advertiser-page-row.jsx

```diff
@@ -41,13 +42,27 @@ const AdvertiserPageRow = ({ row: advert }) => {
         market_rate: effective_rate,
     });
 
-    const showAdForm = () => {
+    const showBuySellForm = () => {
         buy_sell_store.setSelectedAdState(advert);
         showModal({
             key: 'BuySellModal',
         });
     };
 
+    const onBuySellButtonClick = () => {
+        if (general_store.is_advertiser) {
+            showBuySellForm();
+        } else {
+            showModal({
+                key: 'NicknameModal',
+                props: {
+                    onConfirm: showBuySellForm,
+                    should_hide_close_btn: !is_mobile,
```

Comment:
could we use `is_desktop` here instead?

```suggestion
                    should_hide_close_btn: is_desktop,
```